### PR TITLE
Search for libtrufflenfi under a OS-ARCH directory

### PR DESCRIPTION
**Pull request by eregon**
That seems the simplest change for now.
We could also use `source` a file generated by `mx build` but then we might as well move most of the Truffle-related logic like that and it's a larger change.

### CODE REVIEW

**Comment by eregon** [comment]
Alternative to #689.

File: bin/truffleruby.sh [hunk 1]
```diff
@@ -77,7 +99,14 @@ java_args+=("-Xbootclasspath/a:$truffle/dists/truffle-api.jar:$graal_sdk")
 CP="$CP:$truffle/dists/truffle-nfi.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby-launcher.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby.jar"
-java_args+=("-Dtruffle.nfi.library=$truffle/truffle-nfi-native/bin/libtrufflenfi.$(libext)")
+
+libtrufflenfi_os_arch="$truffle/$os-$arch/truffle-nfi-native/bin/libtrufflenfi.$libext"
+if [ -f "$libtrufflenfi_os_arch" ]; then
+  libtrufflenfi="$libtrufflenfi_os_arch"
+else
+  libtrufflenfi="$truffle/truffle-nfi-native/bin/libtrufflenfi.$libext"
```
Comment (pitr-ch) [inline bin/truffleruby.sh]: Please include not-found path https://github.com/graalvm/truffleruby/pull/689/files#diff-851e210aae12b5c6f8feb97ae3ac838aR95 and a todo to remove the fallback https://github.com/graalvm/truffleruby/pull/689/files#diff-851e210aae12b5c6f8feb97ae3ac838aR82

File: bin/truffleruby.sh [hunk 2]
```diff
@@ -12,17 +12,39 @@
...
+    ;;
+  *)
+    echo "unknown platform $(uname)" 1>&2
+    exit 1
+    ;;
+esac
+
+# Check architecture
+case $(uname -m) in
+  x86_64)
```
Comment (pitr-ch) [inline bin/truffleruby.sh]: Can `uname -m` return amd64 on an AMD processor? (Questions like this is the reason why the original PR opted out to reuse mx.)

Comment (eregon) [reply bin/truffleruby.sh]: I have not seen it on Linux/Darwin, if it does if will fail clearly and show the value we do not handle yet.
And it's very easy to add one.

File: bin/truffleruby.sh (hunk 1 above)
Comment (eregon) [reply bin/truffleruby.sh]: I'll add the TODO.
For the not found case, I think the UnsatisifiedLinkError which is caused later is clear enough, but we might as well check earlier, right.

Comment (eregon) [reply bin/truffleruby.sh]: Done.

**Pull request merged**